pydantic_xml/serializers.py

```diff
@@ -558,6 +558,63 @@ class HomogeneousSerializerFactory:
     Homogeneous collection type serializer factory.
     """
 
+    class TextSerializer(Serializer):
+        def __init__(
+                self, model: Type['pxml.BaseXmlModel'], model_field: pd.fields.ModelField, ctx: Serializer.Context,
+        ):
+            assert len(model_field.sub_fields) == 1
+            if isclass(model_field.type_) and issubclass(model_field.type_, pxml.BaseXmlModel):
+                raise errors.ModelFieldError(
+                     model.__name__, model_field.name, "Inline list value should be of scalar type",
+                )
+
+        def serialize(
+                self, element: etree.Element, value: Collection[Any], *, encoder: XmlEncoder, skip_empty: bool = False,
+        ) -> Optional[etree.Element]:
+            if value is None or skip_empty and len(value) == 0:
+                return element
+
+            encoded = " ".join(encoder.encode(val) for val in value)
+            element.text = encoded
+            return element
+
+        def deserialize(self, element: etree.Element) -> Optional[List[Any]]:
+            if element.text is None:
+                return []
```

Comment:
I think it is better to return `None` in that case because it will allow to set a default value if text is `None`. If it returns `[]` default value can't be set.

Reply:
I see what you mean, but I'm thinking of changing the return type to just `List[Any]`. The reason is that this PR is mostly about making pydantic-xml compatible with the `xs:list` xml schema  type. In that case, an empty element said to be a list of ints should return an empty list. Returning `None` would mean, IMO, that the entire element is missing.

I will make the changes if you still want them however.

Reply:
I changed my mind, you are right :)